syntax = "proto3";

import "basic.proto";

package autodrive.workflow.proto.perception_base;

message Rect {
  float left = 1;
  float top = 2;
  float right = 3;
  float bottom = 4;
}

message RectInt {
  int32 left = 1;
  int32 top = 2;
  int32 right = 3;
  int32 bottom = 4;
}

message ImageSpace {
  RectInt rect = 1;
}

message Polygon {
  repeated Vector3f polygon_points = 1;
}

message CameraMatrix {
  repeated float mat_gnd2img = 1;
  repeated float mat_img2gnd = 2;
  repeated float mat_vcsgnd2img = 3;
  repeated float mat_img2vcsgnd = 4;
  repeated float mat_local2img = 5;
  repeated float mat_img2local = 6;
  repeated float mat_vcs2img = 7;
}

message RoiMap {
  Rect model_roi = 1;
  int32 model_id = 2;
}

message Header {
  int32 cam_idx = 1;
  int64 frame_id = 2;
  int64 time_stamp = 3;
}

message Category {
  int32 property_type = 1;
  int32 property = 2;
  float property_conf = 3;
  string property_name = 4;
}

enum ImageId {
  kImageIdCylinderRear = 0;
  kImageIdCylinderFront = 1;
  kImageIdCylinderLeft = 2;
  kImageIdCylinderRight = 3;
  kImageIdIpm = 4;
  kImageIdRear = 5;
  kImageIdFront = 6;
  kImageIdCount = 7;
}

enum RoiId {
  kRoiIdNone = 0;
  kRoiIdObstacle1 = 1;
  kRoiIdObstacle2 = 2;
  kRoiIdLane3 = 3;
  kRoiIdLane4 = 4;
}

enum VehProperty {
  kVehPropertyType = 0;
  kVehPropertyClassification = 1;
  kVehPropertyLight = 2;
  kVehProperty3DBox = 3;
  kVehPropertyKeyPoints = 4;
  kVehPropertyVehBBox = 5;
  kVehPropertyKeyPointsConf = 6;
  kVehPropertyBigSmallCls = 7;
  kVehPropertyPoliceLight = 8;
  kVehPropertyOcclusion = 9;
  kVehPropertyPose = 10;
  kVehPropertyVehFull = 11;
  kVehPropertyOritationCls = 12;
  kVehPropertyOritationRe = 13;
  kVehProperty12KeyPoints = 14;
  kVehProperty12KeyPointsConf = 15;
  kVehproperty12KeyPointsCls = 16;
  kVehPropertyTruncation = 17;
  kVehPropertyCnt = 18;
}

enum PedProperty {
  kPedPropertyOrientation = 0;
  kPedPropertyCyclist = 1;
  kPedPropertyAge = 2;
  kPedPropertyPose = 3;
  kPedPropertyPosNeg = 4;
  kPedPropertyBBox = 5;
  kPedPropertyOcclusion = 6;
  kPedPropertyPosNegOcc = 7;
  kPedProperty2PEDet = 8;
  kPedPropertyCnt = 9;
}

enum VehRearProperty {
  kVehRearPropertyOcclusion = 0;
  kVehRearPropertyPart = 1;
  kVehRearPropertyBBox = 2;
  kVehRearPropertyLmks3 = 3;
  kVehRearPropertyScaleRe = 4;
  kVehRearPropertyPlate = 5;
  kVehRearPropertyCnt = 6;
}

message Box3D {
  Vector3f lower_lt = 1;
  Vector3f lower_lb = 2;
  Vector3f lower_rb = 3;
  Vector3f lower_rt = 4;
  Vector3f upper_lt = 5;
  Vector3f upper_lb = 6;
  Vector3f upper_rb = 7;
  Vector3f upper_rt = 8;
  float conf = 9;
}

enum SceneRegionType {
  kSceneRegionTypeUnknown = 0;
  kIndoor = 1;
  kOutdoor = 2;
}

enum SceneIlluminationType {
  kSceneIlluminationTypeUnknown = 0;
  kDark = 1;
  kBright = 2;
}

enum SceneParkingType {
  kSceneParkingTypeUnknown = 0;
  kSlot = 1;
  kMechanicalSlot = 2;
}

message SceneParsing {
  perception_base.ImageId image_id = 1;
  SceneRegionType region_type = 2;
  SceneIlluminationType illu_type = 3;
  SceneParkingType parking_type = 4;
  float region_conf = 5;
  float illu_conf = 6;
  float parking_conf = 7;
}
